src/app/Shared/Services/Login.service.tsx

```diff
@@ -247,9 +302,13 @@ export class LoginService {
     this.sessionState.next(SessionState.NO_USER_SESSION);
   }
 
-  private navigateToLoginPage(): void {
+  private resetAuthMethod(): void {
     this.authMethod.next(AuthMethod.UNKNOWN);
     this.removeCacheItem(this.AUTH_METHOD_KEY);
+  }
+
+  private navigateToLoginPage(): void {
+    this.resetAuthMethod();
```

Comment:
How about we call `resetAuthMethod` outside `navigateToLoginPage`, just to be clear:

```ts
     tap(() => {
        this.resetSessionState();
        this.resetAuthMethod();
        this.resetAuthMethod();
      }),
```